include/seqan3/alignment/pairwise/alignment_algorithm.hpp

```diff
@@ -644,40 +643,62 @@ class alignment_algorithm :
+     *        range.
+     * \tparam indexed_sequence_pair_range_t The type of the indexed sequence pair range.
+     *
+     * \param[in] index_sequence_pairs The range over indexed sequence pairs.
+     *
+     * \returns A std::vector over seqan3::alignment_result with the requested alignment outcomes.
+     *
+     * \details
+     *
+     * This function is called for the vectorised algorithm. In this case the alignment state stores the results for
+     * the entire chunk of sequence pairs processed within this alignment computation. Accordingly, the chunk of
+     * sequence pairs is processed iteratively and the alignment results are added to the returned vector.
+     * Depending on the selected configuration the following is extracted and/or computed:
+     *
+     * 1. The alignment score.
+     * 2. The end positions of the aligned range for the first and second sequence.
+     * 3. The begin positions of the aligned range for the first and second sequence.
+     * 4. The alignment between both sequences in the respective aligned region.
+     *
+     * If the alignment is run in debug mode (see seqan3::align_cfg::debug) the debug score and optionally trace matrix
+     * are stored in the alignment result as well.
+     */
+    template <typename indexed_sequence_pair_range_t>
     //!\cond
         requires traits_t::is_vectorised
     //!\endcond
-    constexpr auto make_alignment_result(index_range_t && index_range,
-                                         sequence1_range_t && sequence1_range,
-                                         [[maybe_unused]] sequence2_range_t && sequence2_range)
+    constexpr auto make_alignment_result(indexed_sequence_pair_range_t && index_sequence_pairs)
     {
-        using sequence1_t = std::ranges::range_value_t<sequence1_range_t>;
-        using sequence2_t = std::ranges::range_value_t<sequence2_range_t>;
+        using indexed_sequence_pair_t = std::ranges::range_value_t<indexed_sequence_pair_range_t>;
+        using sequence_pair_t = std::tuple_element_t<0, indexed_sequence_pair_t>;
+        using sequence1_t = std::tuple_element_t<0, sequence_pair_t>;
+        using sequence2_t = std::tuple_element_t<1, sequence_pair_t>;
 
         using result_value_t = typename align_result_selector<sequence1_t, sequence2_t, config_t>::type;
 
-        size_t number_of_computed_alignments = std::ranges::distance(sequence1_range);
-        auto index_it = std::ranges::begin(index_range);
-        std::vector<alignment_result<result_value_t>> results{number_of_computed_alignments};
-        // Note we are not using the index range as it might be an infinity range.
-        for (size_t i = 0; i < number_of_computed_alignments; ++i, ++index_it)
+        std::vector<alignment_result<result_value_t>> results{};
+        results.reserve(std::ranges::distance(index_sequence_pairs));
+
+        size_t simd_index = 0;
+        for (auto && [sequence_pairs, alignment_index] : index_sequence_pairs)
```

Comment:
```suggestion
        for (auto && alignment_index : index_sequence_pairs | views::get<1> )
```
? Then you do not need to `(void)` the first tuple element.

Reply:
I will need the sequences for the computation of the alignment. Which is the next step.